File: backend/src/main/java/com/nammametro/simulation/metro/domain/model/Line.java
package com.nammametro.simulation.metro.domain.model;

import java.util.List;

/** {@code orderedStations} is the line's travel-order station sequence, terminus to terminus. */
public record Line(
        long id,
        String code,
        String name,
        String colorHex,
        List<Station> orderedStations
) {
}

File: frontend/src/domain/metro/line.ts
import type { Station } from "./station";

export interface Line {
  readonly id: number;
  readonly code: string;
  readonly name: string;
  readonly colorHex: string;
  readonly stations: readonly Station[];
}

export interface LineSummary {
  readonly id: number;
  readonly code: string;
  readonly name: string;
  readonly colorHex: string;
}
